File: examples/companion-packages/go-gomod-app/internal/pawmatch/flags.go
package pawmatch

import "github.com/agentxm/example-tinyflags/tinyflags"

// Boolean flag names.
const (
	FlagHomeCheckFollowup       = "home-check-followup"
	FlagFeeBreakdownDetailed    = "fee-breakdown-detailed"
	FlagLongStayHighlight       = "long-stay-highlight"
	FlagSuggestDonateAfterAdopt = "suggest-donate-after-adoption"
	FlagShowCharityRatings      = "show-charity-ratings"
	FlagRecommendationStrategy  = "recommendation-strategy"
	FlagMatchQuizDepth          = "match-quiz-depth"
	FlagPetCardStyle            = "pet-card-style"
	FlagDonateFocusDefault      = "donate-focus-default"
)

// NewFlags returns the package-level TinyFlags flag set used by the PawMatch
// CLI. Definitions intentionally mirror the javascript-npm-app port so the
// companion skills see the same seams in every ecosystem.
func NewFlags() *tinyflags.Flags {
	return tinyflags.MustNew(map[string]tinyflags.Flag{
		FlagHomeCheckFollowup: tinyflags.MustBooleanFlag(
			tinyflags.BoolDefault(false),
			tinyflags.BoolRollout(25),
		),
		FlagFeeBreakdownDetailed: tinyflags.MustBooleanFlag(
			tinyflags.BoolDefault(true),
		),
		FlagLongStayHighlight: tinyflags.MustBooleanFlag(
			tinyflags.BoolDefault(true),
		),
		FlagSuggestDonateAfterAdopt: tinyflags.MustBooleanFlag(
			tinyflags.BoolDefault(false),
			tinyflags.BoolRollout(50),
		),
		FlagShowCharityRatings: tinyflags.MustBooleanFlag(
			tinyflags.BoolDefault(true),
		),
		FlagRecommendationStrategy: tinyflags.MustVariantFlag(
			[]string{"popularity", "match-quiz", "longest-stay"},
			tinyflags.VariantDefault("match-quiz"),
			tinyflags.VariantRollout(map[string]int{"longest-stay": 20}),
		),
		FlagMatchQuizDepth: tinyflags.MustVariantFlag(
			[]string{"short", "standard", "thorough"},
			tinyflags.VariantDefault("standard"),
		),
		FlagPetCardStyle: tinyflags.MustVariantFlag(
			[]string{"compact", "detailed", "playful"},
			tinyflags.VariantDefault("detailed"),
		),
		FlagDonateFocusDefault: tinyflags.MustVariantFlag(
			[]string{"all", "shelters", "rescue"},
			tinyflags.VariantDefault("all"),
		),
	})
}

File: examples/companion-packages/swift-cocoapods-app/PawMatch/Flags.swift
// Flag definitions and constants for the PawMatch CLI.

import AgentXMExampleTinyFlags

enum PawMatchFlag {
    static let homeCheckFollowup = "home-check-followup"
    static let feeBreakdownDetailed = "fee-breakdown-detailed"
    static let longStayHighlight = "long-stay-highlight"
    static let suggestDonateAfterAdoption = "suggest-donate-after-adoption"
    static let showCharityRatings = "show-charity-ratings"
    static let recommendationStrategy = "recommendation-strategy"
    static let matchQuizDepth = "match-quiz-depth"
    static let petCardStyle = "pet-card-style"
    static let donateFocusDefault = "donate-focus-default"

    /// Build the TinyFlags bundle the CLI evaluates at runtime. Every flag
    /// declared here is wired into at least one command path so the companion
    /// skills have realistic targets.
    static func makeTinyFlags() throws -> TinyFlags {
        try TinyFlags.builder()
            .boolean(homeCheckFollowup, default: false, rollout: 25)
            .boolean(feeBreakdownDetailed, default: true)
            .boolean(longStayHighlight, default: true)
            .boolean(suggestDonateAfterAdoption, default: false, rollout: 50)
            .boolean(showCharityRatings, default: true)
            .variant(
                recommendationStrategy,
                variants: ["popularity", "match-quiz", "longest-stay"],
                default: "match-quiz",
                rollout: ["longest-stay": 20]
            )
            .variant(
                matchQuizDepth,
                variants: ["short", "standard", "thorough"],
                default: "standard"
            )
            .variant(
                petCardStyle,
                variants: ["compact", "detailed", "playful"],
                default: "detailed"
            )
            .variant(
                donateFocusDefault,
                variants: ["all", "shelters", "rescue"],
                default: "all"
            )
            .build()
    }
}
